plugins/sf2_player/sf2_player.cpp

```diff
@@ -267,6 +268,46 @@ void sf2Instrument::loadFile( const QString & _file )
 		// soundfont for the first time fails
 		updateSampleRate();
 	}
+
+	// setting the first bank and patch number that is found
+	bool foundFirstPreset = false;
+	auto sSoundCount = ::fluid_synth_sfcount( m_synth );
+	for ( int i = 0; i < sSoundCount; ++i ) {
+		int iBank = 0;
+		int iProg = 0;
+		fluid_sfont_t *pSoundFont = ::fluid_synth_get_sfont( m_synth, i );
+
+		if ( pSoundFont ) {
+#ifdef CONFIG_FLUID_BANK_OFFSET
+			int iBankOff = ::fluid_synth_get_bank_offset( m_synth, fluid_sfont_get_id( pSoundFont ) );
+#endif
+
+			fluid_sfont_iteration_start( pSoundFont );
+#if FLUIDSYNTH_VERSION_MAJOR < 2
+			fluid_preset_t preset;
+			fluid_preset_t *pCurPreset = &preset;
+#else
+			fluid_preset_t *pCurPreset;
+#endif
+
+			while ( ( pCurPreset = fluid_sfont_iteration_next_wrapper( pSoundFont, pCurPreset ) ) ) {
+#ifdef CONFIG_FLUID_BANK_OFFSET
+				iBank += iBankOff;
+#endif
+				if ( pCurPreset && !foundFirstPreset ) {
```

Comment:
* Checking for `pCurPreset` looks redundant because the `while` statement already checks for it.
* Can you use an `if` statement and a `break` statement instead of this double-loop? The current implementation looks very inefficient.

Reply:
Made the fix https://github.com/LMMS/lmms/pull/5084/commits/37e151b5fc1e7661d153a782e449e6db91227257